lnrpc/routerrpc/router_server.go

```diff
@@ -322,7 +324,7 @@ func marshallError(sendError error) (*Failure, error) {
 
 	switch onionErr := fErr.FailureMessage.(type) {
 
-	case *lnwire.FailUnknownPaymentHash:
+	case *lnwire.FailIncorrectDetails:
```

Comment:
at some point we may want to consider renaming/changing the rpc level error, since this now encompasses more than just unknown payment hash

Reply:
Added that rename too. 

I also noticed that the assigned integers for the failure codes were not matching the spec, even though that was the original intention. Added commit to update the comment.